commands/list_keypairs_test.go

```diff
@@ -3,75 +3,146 @@ package commands
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`[]`))
+	}))
+	defer keyPairsMockServer.Close()
+
 	// needed to prevent search for the default cluster
-	cmdClusterID = "foobar"
-	err := checkListKeypairs(ListKeypairsCommand, []string{})
+	args := listKeypairsArguments{}
+	args.apiEndpoint = keyPairsMockServer.URL
+	args.token = "my-token"
+	args.clusterID = "my-cluster"
+
+	err := listKeypairsValidate(&args)
 	if err != nil {
-		t.Error("Login token not accepted:", err)
+		t.Error(err)
+	}
+
+	result, listErr := listKeypairs(args)
+	if listErr != nil {
+		t.Error(listErr)
+	}
+	if len(result.keypairs) > 0 {
+		t.Error("Got key pairs where we expected none.")
 	}
 }
 
-// Not logged in, but a cluster ID is given. Should be okay.
-func Test_CheckListKeypairs_NotLoggedInWithCluster(t *testing.T) {
-	cmdClusterID = "foobar"
-	err := checkListKeypairs(ListKeypairsCommand, []string{})
+// Test_ListKeypairs_NotFound simulates the situation where the cluster
+// to list key pairs for is not found
+func Test_ListKeypairs_NotFound(t *testing.T) {
+	defer viper.Reset()
+	dir := tempDir()
+	defer os.RemoveAll(dir)
+	config.Initialize(dir)
+
+	keyPairsMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte(`{"code": "RESOURCE_NOT_FOUND", "message": "The cluster could not be found."}`))
+	}))
+	defer keyPairsMockServer.Close()
+
+	args := listKeypairsArguments{}
+	args.apiEndpoint = keyPairsMockServer.URL
+	args.token = "my-token"
+	args.clusterID = "unknown-cluster"
+
+	err := listKeypairsValidate(&args)
 	if err != nil {
 		t.Error(err)
 	}
-}
 
-// Not logged in, no cluster ID is given. Should fail.
-func Test_CheckListKeypairs_NotLoggedInNoCluster(t *testing.T) {
-	cmdClusterID = ""
-	err := checkListKeypairs(ListKeypairsCommand, []string{})
-	if err == nil {
-		t.Error("Expected error didn't happen:", err)
+	_, listErr := listKeypairs(args)
+	if listErr == nil {
+		t.Error("No error occurred where we expected one.")
+	} else if listErr.Error() != errClusterNotFound {
+		t.Errorf("Expected error '%s', got '%s'.", errClusterNotFound, listErr.Error())
 	}
 }
 
-func Test_ListKeyPairs(t *testing.T) {
-	// mock service returning key pairs
+// Test_ListKeyPairs_Nonempty simulates listing key pairs where several
+// items are returned.
+func Test_ListKeyPairs_Nonempty(t *testing.T) {
+	defer viper.Reset()
+	dir := tempDir()
+	defer os.RemoveAll(dir)
+	config.Initialize(dir)
+
+	// mock service returning key pairs. For the sake of simplicity,
+	// it doesn't care about auth tokens.
 	keyPairsMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusOK)
 		w.Write([]byte(`[
+			{
```

Comment:
It is a spaces vs tabs thing. Here you use tabs. The other code in the string uses spaces. Atom renders it the same way depending on your settings. Github displays spaces and tabs using different widths for this reason, to notice you used different characters. So please change it to spaces.